untyped-lambda-calculus/article.md

```diff
@@ -0,0 +1,461 @@
+# Untyped Lambda Calculus
+
+When talking about Haskell, the term "lambda calculus" often crops up.
+It is a theoretical framework used to define the meaning of "computation" in many functional languages.
+Understanding lambda calculus can be very helpful when talking about Haskell programs.
+It is also used extensively in programming language theory research and functional language design.
+
+There are two broad kinds of lambda calculus, _typed_, and _untyped_.
+The _untyped_ kind, discussed in this article, expresses an unrestricted computation.
+However, as a system of logic, it is self-contradictory.
+The _typed_ kind, which we intend to discuss in a follow-up article, is a restricted version.
+It doesn't have the logical paradox that plagues the untyped kind and can serve as a logic system.
+
+The list of languages based on lambda calculus includes:
+
+- Haskell
+- Most languages in the ML family, like OCaml and F#
```

Comment:
Haskell is also in the ML family, no? Somewhat odd phrasing.

Reply:
Haskell doesn't belong to the ML family

Reply:
Fair enough. 😅